ValveResourceFormat/IO/GameFileLoader.cs

```diff
@@ -0,0 +1,482 @@
+
+            if (CurrentPackage != null && CurrentPackage.Entries.TryGetValue("vpk", out var vpkEntries))
+            {
+                foreach (var searchPath in vpkEntries)
+                {
+                    if (!CachedPackages.TryGetValue(searchPath.GetFileName(), out var package))
+                    {
+                        Console.WriteLine($"Preloading vpk \"{searchPath.GetFullPath()}\" from parent vpk");
+
+                        var stream = GetPackageEntryStream(CurrentPackage, searchPath);
+                        package = new Package();
+                        package.OptimizeEntriesForBinarySearch(StringComparison.OrdinalIgnoreCase);
+                        package.SetFileName(searchPath.GetFileName());
+                        package.Read(stream);
+                        CachedPackages[searchPath.GetFileName()] = package;
+                    }
+
+                    packages.Add(package);
+                }
+            }
+
+            foreach (var package in packages)
+            {
+                entry = package?.FindEntry(file);
+
+                if (entry != null)
+                {
+#if DEBUG_FILE_LOAD
+                    Console.WriteLine($"Loaded \"{file}\" from preloaded vpk \"{package.FileName}\"");
+#endif
+
+                    return (null, package, entry);
+                }
+            }
+
+            var path = FindResourcePath(paths.Concat(CurrentGameSearchPaths).ToList(), file, FileName);
+
+            if (path != null)
+            {
+                return (path, null, null);
+            }
+
+            if (logNotFound)
+            {
+                Console.Error.WriteLine($"Failed to load \"{file}\". Did you configure VPK paths in settings correctly?");
+            }
+
+            if (string.IsNullOrEmpty(file) || file == "_c")
+            {
+                Console.Error.WriteLine($"Empty string passed to file loader here: {Environment.StackTrace}");
+
+#if DEBUG_FILE_LOAD
+                System.Diagnostics.Debugger.Break();
+#endif
+            }
+
+            return (null, null, null);
+        }
+
+        public virtual Resource LoadFile(string file)
+        {
+            Resource resource = new Resource
+            {
+                FileName = file,
+            };
+
+            var foundFile = FindFile(file);
+
+            if (foundFile.PathOnDisk != null)
+            {
+                resource.Read(foundFile.PathOnDisk);
+                return resource;
+            }
+            else if (foundFile.PackageEntry != null)
+            {
+                var stream = GetPackageEntryStream(foundFile.Package, foundFile.PackageEntry);
+                resource.Read(stream);
+                return resource;
+            }
+
+            return null;
+        }
+
+        public ShaderCollection LoadShader(string shaderName) => null;
+
+        public static Stream GetPackageEntryStream(Package package, PackageEntry entry)
```

Comment:
Let's not move this into lib code. This should be done in https://github.com/ValveResourceFormat/ValvePak/issues/10

Reply:
oh, where did that come from, my bad will fix

Reply:
hmmm seems like it is needed by IFileLoader?